subsys/net/lib/wifi_mgmt_ext/wifi_mgmt_ext.c

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <net/wifi_mgmt_ext.h>
+#include <net/wifi_credentials.h>
+
+#include <zephyr/kernel.h>
+#include <zephyr/net/net_core.h>
+#include <zephyr/net/net_if.h>
+#include <zephyr/net/wifi_mgmt.h>
+
+#include "wpa_cli_zephyr.h"
+
+/* Ensure 'strnlen' is available even with -std=c99. */
+size_t strnlen(const char *buf, size_t bufsz);
+
+LOG_MODULE_REGISTER(wifi_mgmt_ext, CONFIG_WIFI_MGMT_EXT_LOG_LEVEL);
+
+static inline const char *wpa_supp_security_txt(enum wifi_security_type security)
+{
+	switch (security) {
+	case WIFI_SECURITY_TYPE_NONE:
+		return "OPEN";
+	case WIFI_SECURITY_TYPE_PSK:
+		return "WPA-PSK";
+	case WIFI_SECURITY_TYPE_PSK_SHA256:
+		return "WPA-PSK-SHA256";
+	case WIFI_SECURITY_TYPE_SAE:
+		return "SAE";
+	case WIFI_SECURITY_TYPE_UNKNOWN:
+	default:
+		return "UNKNOWN";
+	}
+}
+
+static int add_network_from_credentials_struct_personal(struct wifi_credentials_personal *creds)
+{
+	int ret = 0;
+	struct add_network_resp resp = {0};
+
+	ret = z_wpa_ctrl_add_network(&resp);
+	if (ret) {
+		LOG_ERR("Encoding error: %d returned at %s:%d", ret, __FILE__, __LINE__);
+		return ret;
+	}
+
+	z_wpa_cli_cmd_v("set_network %d ssid \"%s\"", resp.network_id, creds->header.ssid);
```

Comment:
I have a PR that adds a macro to check return value https://github.com/nrfconnect/sdk-hostap/pull/59 (though the macro is disallowed by Zephyr checkpatch). This is mainly for memory allocation/socket related failures.

Reply:
Would be nice, but if it doesn't go through the compliance check, we cannot add it.